mongo/mongologger/mongologger.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongologger // import "go.mongodb.org/mongo-driver/mongo/mongologger"
+
+import (
+	"os"
+	"strconv"
+)
+
+// Logger is an interface for loggers to implement
+type Logger interface {
+	Trace(message string, args ...Field)
+	Debug(message string, args ...Field)
+	Info(message string, args ...Field)
+	Notice(message string, args ...Field)
+	Warning(message string, args ...Field)
+	Error(message string, args ...Field)
+}
+
+// MongoLogger manages the mongo log messages sent to the logger
+type MongoLogger struct {
+	logger Logger
+
+	logFullCommands      bool
+	commandLevel         Level
+	connectionLevel      Level
+	sdamLevel            Level
+	serverSelectionLevel Level
+}
+
+// NewMongoLogger creates a new Mongologger to use with a mongo.Client. Log level defaults to Warning
+// for all components.
+func NewMongoLogger(opts ...*Options) (*MongoLogger, error) {
+	mlo := MergeOptions(opts...)
+
+	if mlo.Logger == nil {
+		dl := &defaultLogger{}
+
+		dl.writer = os.Stderr
+		var err error
+		if mlo.OutputFile != nil {
+			dl.outputFile = *mlo.OutputFile
+			dl.writer, err = os.Create(dl.outputFile)
```

Comment:
Does the spec say how the file should be opened (e..g what mode, what to do if the file already exists, etc)?

Reply:
I'd recommend bringing this up in the spec so it's standardized across drivers.

Reply:
brought it up on the spec